// auth0/dist/cjs/management/__generated/models/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetActions200ResponseActionsInnerSupportedTriggersInnerIdAnyOf = exports.GetActions200ResponseActionsInnerIntegrationCurrentReleaseRequiredSecretsInnerTypeEnum = exports.GetActions200ResponseActionsInnerIntegrationFeatureTypeEnum = exports.GetActions200ResponseActionsInnerStatusEnum = exports.GetActionVersions200ResponseVersionsInnerStatusEnum = exports.FactorNameEnum = exports.EnrollmentStatusEnum = exports.EmailTemplateUpdateTemplateEnum = exports.EmailProviderUpdateNameEnum = exports.EmailProviderCreateNameEnum = exports.DeviceCredentialCreateTypeEnum = exports.DeviceCredentialTypeEnum = exports.CustomDomainTypeEnum = exports.CustomDomainStatusEnum = exports.ConnectionUpdateOptionsSetUserRootAttributesEnum = exports.ConnectionUpdateOptionsPasswordPolicyEnum = exports.ConnectionCreateOptionsPasskeyOptionsChallengeUiEnum = exports.ConnectionCreateOptionsSetUserRootAttributesEnum = exports.ConnectionCreateOptionsPasswordPolicyEnum = exports.ConnectionCreateStrategyEnum = exports.ClientUpdateOidcLogoutBackchannelLogoutInitiatorsSelectedInitiatorsEnum = exports.ClientUpdateOidcLogoutBackchannelLogoutInitiatorsModeEnum = exports.ClientUpdateJwtConfigurationAlgEnum = exports.ClientUpdateComplianceLevelEnum = exports.ClientUpdateOrganizationRequireBehaviorEnum = exports.ClientUpdateOrganizationUsageEnum = exports.ClientUpdateAppTypeEnum = exports.ClientUpdateTokenEndpointAuthMethodEnum = exports.ClientRefreshTokenExpirationTypeEnum = exports.ClientRefreshTokenRotationTypeEnum = exports.ClientOidcLogoutBackchannelLogoutInitiatorsSelectedInitiatorsEnum = exports.ClientOidcLogoutBackchannelLogoutInitiatorsModeEnum = exports.ClientJwtConfigurationAlgEnum = exports.ClientGrantCreateOrganizationUsageEnum = exports.ClientGrantOrganizationUsageEnum = exports.ClientDefaultOrganizationFlowsEnum = exports.ClientCreateJwtConfigurationAlgEnum = exports.ClientCreateClientAuthenticationMethodsTlsClientAuthCredentialsInnerCredentialTypeEnum = exports.ClientCreateClientAuthenticationMethodsSelfSignedTlsClientAuthCredentialsInnerCredentialTypeEnum = exports.ClientCreateClientAuthenticationMethodsPrivateKeyJwtCredentialsInnerAlgEnum = exports.ClientCreateClientAuthenticationMethodsPrivateKeyJwtCredentialsInnerCredentialTypeEnum = exports.ClientCreateComplianceLevelEnum = exports.ClientCreateOrganizationRequireBehaviorEnum = exports.ClientCreateOrganizationUsageEnum = exports.ClientCreateAppTypeEnum = exports.ClientCreateTokenEndpointAuthMethodEnum = exports.ClientComplianceLevelEnum = exports.ClientOrganizationRequireBehaviorEnum = exports.ClientOrganizationUsageEnum = exports.ClientTokenEndpointAuthMethodEnum = void 0;
exports.PatchClientGrantsByIdRequestOrganizationUsageEnum = exports.PatchBruteForceProtectionRequestModeEnum = exports.PatchBruteForceProtectionRequestShieldsEnum = exports.PatchBreachedPasswordDetectionRequestStagePreUserRegistrationShieldsEnum = exports.PatchBreachedPasswordDetectionRequestMethodEnum = exports.PatchBreachedPasswordDetectionRequestAdminNotificationFrequencyEnum = exports.PatchBreachedPasswordDetectionRequestShieldsEnum = exports.PatchBindingsRequestBindingsInnerOneOfRefTypeEnum = exports.PatchAuthenticationMethodsByAuthenticationMethodIdRequestPreferredAuthenticationMethodEnum = exports.JobFormatEnum = exports.HookCreateTriggerIdEnum = exports.GetSuspiciousIpThrottling200ResponseShieldsEnum = exports.GetPnProviders200ResponseProviderEnum = exports.GetPhoneProviders200ResponseProviderEnum = exports.GetMessageTypes200ResponseMessageTypesEnum = exports.GetLogStreams200ResponseInnerOneOfSinkHttpContentFormatEnum = exports.GetLogStreams200ResponseInnerOneOfFiltersInnerNameEnum = exports.GetLogStreams200ResponseInnerOneOfFiltersInnerTypeEnum = exports.GetLogStreams200ResponseInnerOneOf7SinkMixpanelRegionEnum = exports.GetLogStreams200ResponseInnerOneOf7TypeEnum = exports.GetLogStreams200ResponseInnerOneOf7StatusEnum = exports.GetLogStreams200ResponseInnerOneOf6TypeEnum = exports.GetLogStreams200ResponseInnerOneOf6StatusEnum = exports.GetLogStreams200ResponseInnerOneOf5TypeEnum = exports.GetLogStreams200ResponseInnerOneOf5StatusEnum = exports.GetLogStreams200ResponseInnerOneOf4TypeEnum = exports.GetLogStreams200ResponseInnerOneOf4StatusEnum = exports.GetLogStreams200ResponseInnerOneOf3SinkDatadogRegionEnum = exports.GetLogStreams200ResponseInnerOneOf3TypeEnum = exports.GetLogStreams200ResponseInnerOneOf3StatusEnum = exports.GetLogStreams200ResponseInnerOneOf2SinkAzureRegionEnum = exports.GetLogStreams200ResponseInnerOneOf2TypeEnum = exports.GetLogStreams200ResponseInnerOneOf2StatusEnum = exports.GetLogStreams200ResponseInnerOneOf1SinkAwsRegionEnum = exports.GetLogStreams200ResponseInnerOneOf1TypeEnum = exports.GetLogStreams200ResponseInnerOneOf1StatusEnum = exports.GetLogStreams200ResponseInnerOneOfTypeEnum = exports.GetLogStreams200ResponseInnerOneOfStatusEnum = exports.GetExecution200ResponseStatusEnum = exports.GetEmailTemplatesByTemplateName200ResponseTemplateEnum = exports.GetCredentials200ResponseInnerAlgEnum = exports.GetBruteForceProtection200ResponseModeEnum = exports.GetBruteForceProtection200ResponseShieldsEnum = exports.GetBreachedPasswordDetection200ResponseStagePreUserRegistrationShieldsEnum = exports.GetBreachedPasswordDetection200ResponseMethodEnum = exports.GetBreachedPasswordDetection200ResponseAdminNotificationFrequencyEnum = exports.GetBreachedPasswordDetection200ResponseShieldsEnum = exports.GetAuthenticationMethods200ResponseOneOfInnerAuthenticationMethodsInnerTypeEnum = exports.GetAuthenticationMethods200ResponseOneOfInnerPreferredAuthenticationMethodEnum = exports.GetAuthenticationMethods200ResponseOneOfInnerTypeEnum = void 0;
exports.PostFlowsVaultConnectionsRequestAnyOf2SetupTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf20AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf2AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf1SetupTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf19AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf18SetupTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf18AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf17AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf16SetupAnyOf1EnvironmentEnum = exports.PostFlowsVaultConnectionsRequestAnyOf16SetupAnyOf1TypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf16SetupAnyOfTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf16AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf15SetupAnyOfTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf15AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf14AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf13AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf12SetupAnyOfTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf12AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf11SetupTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf11AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf10AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf1AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOfAppIdEnum = exports.PostEmailTemplatesRequestTemplateEnum = exports.PostCustomDomainsRequestCustomClientIpHeaderEnum = exports.PostCustomDomainsRequestTlsPolicyEnum = exports.PostCustomDomainsRequestVerificationMethodEnum = exports.PostCustomDomainsRequestTypeEnum = exports.PostCustomDomains201ResponseVerificationMethodsInnerNameEnum = exports.PostCustomDomains201ResponseTypeEnum = exports.PostCustomDomains201ResponseStatusEnum = exports.PostCredentialsRequestCredentialTypeEnum = exports.PostBrandingThemeRequestWidgetSocialButtonsLayoutEnum = exports.PostBrandingThemeRequestWidgetLogoPositionEnum = exports.PostBrandingThemeRequestWidgetHeaderTextAlignmentEnum = exports.PostBrandingThemeRequestPageBackgroundPageLayoutEnum = exports.PostBrandingThemeRequestFontsLinksStyleEnum = exports.PostBrandingThemeRequestBordersInputsStyleEnum = exports.PostBrandingThemeRequestBordersButtonsStyleEnum = exports.PostAuthenticationMethodsRequestPreferredAuthenticationMethodEnum = exports.PostAuthenticationMethodsRequestTypeEnum = exports.PostAuthenticationMethods201ResponsePreferredAuthenticationMethodEnum = exports.PostAuthenticationMethods201ResponseTypeEnum = exports.PatchSuspiciousIpThrottlingRequestShieldsEnum = exports.PatchLogStreamsByIdRequestSinkOneOf3MixpanelRegionEnum = exports.PatchLogStreamsByIdRequestSinkOneOfDatadogRegionEnum = exports.PatchLogStreamsByIdRequestStatusEnum = exports.PatchEmailTemplatesByTemplateNameRequestTemplateEnum = exports.PatchCustomDomainsByIdRequestCustomClientIpHeaderEnum = exports.PatchCustomDomainsByIdRequestTlsPolicyEnum = void 0;
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf6TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf5TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf4TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf3TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf2ConfigPositionEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf2TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf1TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOfTypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf9TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf8TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf7ConfigCategoriesEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf7TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf6TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf5TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf4TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf3TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf2TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf15TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf14TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf13TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf12TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11ConfigChargeAnyOfOneOffCurrencyEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11ConfigChargeAnyOf1TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11ConfigChargeAnyOfTypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11ConfigProviderEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf10ConfigHashEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf10TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf1TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOfTypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf1AnyOf1TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf1AnyOfTypeEnum = exports.PostFormsRequestNodesInnerAnyOf2TypeEnum = exports.PostFormsRequestNodesInnerAnyOf1TypeEnum = exports.PostFormsRequestNodesInnerAnyOfTypeEnum = exports.PostFormsRequestEndingResumeFlowEnum = exports.PostFlowsVaultConnectionsRequestAnyOfSetupTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf9SetupAlgorithmEnum = exports.PostFlowsVaultConnectionsRequestAnyOf9SetupTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf9AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf8AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf7SetupTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf7AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf6AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf5SetupTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf5AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf4SetupTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf4AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf3SetupTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf3AppIdEnum = void 0;
exports.GetDeviceCredentialsTypeEnum = exports.GetConnectionsStrategyEnum = exports.UserEnrollmentAuthMethodEnum = exports.UserEnrollmentStatusEnum = exports.TenantSettingsUpdateFlagsChangePwdFlowV1Enum = exports.TenantSettingsUpdateDeviceFlowCharsetEnum = exports.TenantSettingsUpdateEnabledLocalesEnum = exports.TenantSettingsSessionCookieModeEnum = exports.TenantSettingsDeviceFlowCharsetEnum = exports.TenantSettingsEnabledLocalesEnum = exports.SsProfileUpdateAllowedStrategiesEnum = exports.SsProfileCreateAllowedStrategiesEnum = exports.SsProfileAllowedStrategiesEnum = exports.ResourceServerUpdateConsentPolicyEnum = exports.ResourceServerUpdateTokenDialectEnum = exports.ResourceServerUpdateSigningAlgEnum = exports.ResourceServerTokenEncryptionEncryptionKeyAlgEnum = exports.ResourceServerTokenEncryptionFormatEnum = exports.ResourceServerProofOfPossessionMechanismEnum = exports.ResourceServerCreateConsentPolicyEnum = exports.ResourceServerCreateTokenDialectEnum = exports.ResourceServerCreateSigningAlgEnum = exports.ResourceServerConsentPolicyEnum = exports.ResourceServerTokenDialectEnum = exports.ResourceServerSigningAlgEnum = exports.PutAuthenticationMethodsRequestInnerPreferredAuthenticationMethodEnum = exports.PutAuthenticationMethodsRequestInnerTypeEnum = exports.PutAuthenticationMethods200ResponseInnerAuthenticationMethodsInnerTypeEnum = exports.PutAuthenticationMethods200ResponseInnerPreferredAuthenticationMethodEnum = exports.PutAuthenticationMethods200ResponseInnerTypeEnum = exports.PromptsSettingsUpdateUniversalLoginExperienceEnum = exports.PromptsSettingsUniversalLoginExperienceEnum = exports.PostVerify200ResponseTypeEnum = exports.PostVerify200ResponseStatusEnum = exports.PostVerificationEmailRequestIdentityProviderEnum = exports.PostUsersExportsRequestFormatEnum = exports.PostLogStreamsRequestOneOfFiltersInnerNameEnum = exports.PostLogStreamsRequestOneOfFiltersInnerTypeEnum = exports.PostLogStreamsRequestOneOf7TypeEnum = exports.PostLogStreamsRequestOneOf6TypeEnum = exports.PostLogStreamsRequestOneOf5TypeEnum = exports.PostLogStreamsRequestOneOf4TypeEnum = exports.PostLogStreamsRequestOneOf3TypeEnum = exports.PostLogStreamsRequestOneOf2SinkAzureRegionEnum = exports.PostLogStreamsRequestOneOf2TypeEnum = exports.PostLogStreamsRequestOneOf1SinkAwsRegionEnum = exports.PostLogStreamsRequestOneOf1TypeEnum = exports.PostLogStreamsRequestOneOfTypeEnum = exports.PostIdentitiesRequestProviderEnum = exports.PostFormsRequestNodesInnerAnyOfConfigNextNodeAnyOf = void 0;
exports.GetUsersSearchEngineEnum = exports.DeleteUserIdentityByUserIdProviderEnum = exports.DeleteMultifactorByProviderProviderEnum = exports.PutSelfServiceProfileCustomTextPageEnum = exports.PutSelfServiceProfileCustomTextLanguageEnum = exports.GetSelfServiceProfileCustomTextPageEnum = exports.GetSelfServiceProfileCustomTextLanguageEnum = exports.PutPartialsPromptEnum = exports.PutCustomTextByLanguageLanguageEnum = exports.PutCustomTextByLanguagePromptEnum = exports.GetPartialsPromptEnum = exports.GetCustomTextByLanguageLanguageEnum = exports.GetCustomTextByLanguagePromptEnum = exports.GetHooksTriggerIdEnum = exports.PutFactorsByNameOperationNameEnum = exports.GetFormsByIdHydrateEnum = exports.GetFormsHydrateEnum = exports.GetFlowsExecutionsByExecutionIdHydrateEnum = exports.GetFlowsByIdHydrateEnum = exports.GetFlowsHydrateEnum = exports.PutEmailTemplatesByTemplateNameTemplateNameEnum = exports.PatchEmailTemplatesByTemplateNameOperationTemplateNameEnum = exports.GetEmailTemplatesByTemplateNameTemplateNameEnum = void 0;
exports.ClientTokenEndpointAuthMethodEnum = {
    none: 'none',
    client_secret_post: 'client_secret_post',
    client_secret_basic: 'client_secret_basic',
};
exports.ClientOrganizationUsageEnum = {
    deny: 'deny',
    allow: 'allow',
    require: 'require',
};
exports.ClientOrganizationRequireBehaviorEnum = {
    no_prompt: 'no_prompt',
    pre_login_prompt: 'pre_login_prompt',
    post_login_prompt: 'post_login_prompt',
};
exports.ClientComplianceLevelEnum = {
    none: 'none',
    fapi1_adv_pkj_par: 'fapi1_adv_pkj_par',
    fapi1_adv_mtls_par: 'fapi1_adv_mtls_par',
    null: 'null',
};
exports.ClientCreateTokenEndpointAuthMethodEnum = {
    none: 'none',
    client_secret_post: 'client_secret_post',
    client_secret_basic: 'client_secret_basic',
};
exports.ClientCreateAppTypeEnum = {
    native: 'native',
    spa: 'spa',
    regular_web: 'regular_web',
    non_interactive: 'non_interactive',
    rms: 'rms',
    box: 'box',
    cloudbees: 'cloudbees',
    concur: 'concur',
    dropbox: 'dropbox',
    mscrm: 'mscrm',
    echosign: 'echosign',
    egnyte: 'egnyte',
    newrelic: 'newrelic',
    office365: 'office365',
    salesforce: 'salesforce',
    sentry: 'sentry',
    sharepoint: 'sharepoint',
    slack: 'slack',
    springcm: 'springcm',
    zendesk: 'zendesk',
    zoom: 'zoom',
    sso_integration: 'sso_integration',
    oag: 'oag',
};
exports.ClientCreateOrganizationUsageEnum = {
    deny: 'deny',
    allow: 'allow',
    require: 'require',
};
exports.ClientCreateOrganizationRequireBehaviorEnum = {
    no_prompt: 'no_prompt',
    pre_login_prompt: 'pre_login_prompt',
    post_login_prompt: 'post_login_prompt',
};
exports.ClientCreateComplianceLevelEnum = {
    none: 'none',
    fapi1_adv_pkj_par: 'fapi1_adv_pkj_par',
    fapi1_adv_mtls_par: 'fapi1_adv_mtls_par',
    null: 'null',
};
exports.ClientCreateClientAuthenticationMethodsPrivateKeyJwtCredentialsInnerCredentialTypeEnum = {
    public_key: 'public_key',
};
exports.ClientCreateClientAuthenticationMethodsPrivateKeyJwtCredentialsInnerAlgEnum = {
    RS256: 'RS256',
    RS384: 'RS384',
    PS256: 'PS256',
};
exports.ClientCreateClientAuthenticationMethodsSelfSignedTlsClientAuthCredentialsInnerCredentialTypeEnum = {
    x509_cert: 'x509_cert',
};
exports.ClientCreateClientAuthenticationMethodsTlsClientAuthCredentialsInnerCredentialTypeEnum = {
    cert_subject_dn: 'cert_subject_dn',
};
exports.ClientCreateJwtConfigurationAlgEnum = {
    HS256: 'HS256',
    RS256: 'RS256',
    PS256: 'PS256',
};
exports.ClientDefaultOrganizationFlowsEnum = {
    client_credentials: 'client_credentials',
};
exports.ClientGrantOrganizationUsageEnum = {
    deny: 'deny',
    allow: 'allow',
    require: 'require',
};
exports.ClientGrantCreateOrganizationUsageEnum = {
    deny: 'deny',
    allow: 'allow',
    require: 'require',
};
exports.ClientJwtConfigurationAlgEnum = {
    HS256: 'HS256',
    RS256: 'RS256',
    PS256: 'PS256',
};
exports.ClientOidcLogoutBackchannelLogoutInitiatorsModeEnum = {
    custom: 'custom',
    all: 'all',
};
exports.ClientOidcLogoutBackchannelLogoutInitiatorsSelectedInitiatorsEnum = {
    rp_logout: 'rp-logout',
    idp_logout: 'idp-logout',
    password_changed: 'password-changed',
    session_expired: 'session-expired',
    session_revoked: 'session-revoked',
    account_deleted: 'account-deleted',
    email_identifier_changed: 'email-identifier-changed',
    mfa_phone_unenrolled: 'mfa-phone-unenrolled',
    account_deactivated: 'account-deactivated',
};
exports.ClientRefreshTokenRotationTypeEnum = {
    rotating: 'rotating',
    non_rotating: 'non-rotating',
};
exports.ClientRefreshTokenExpirationTypeEnum = {
    expiring: 'expiring',
    non_expiring: 'non-expiring',
};
exports.ClientUpdateTokenEndpointAuthMethodEnum = {
    none: 'none',
    client_secret_post: 'client_secret_post',
    client_secret_basic: 'client_secret_basic',
    null: 'null',
};
exports.ClientUpdateAppTypeEnum = {
    native: 'native',
    spa: 'spa',
    regular_web: 'regular_web',
    non_interactive: 'non_interactive',
    rms: 'rms',
    box: 'box',
    cloudbees: 'cloudbees',
    concur: 'concur',
    dropbox: 'dropbox',
    mscrm: 'mscrm',
    echosign: 'echosign',
    egnyte: 'egnyte',
    newrelic: 'newrelic',
    office365: 'office365',
    salesforce: 'salesforce',
    sentry: 'sentry',
    sharepoint: 'sharepoint',
    slack: 'slack',
    springcm: 'springcm',
    zendesk: 'zendesk',
    zoom: 'zoom',
    sso_integration: 'sso_integration',
    oag: 'oag',
};
exports.ClientUpdateOrganizationUsageEnum = {
    deny: 'deny',
    allow: 'allow',
    require: 'require',
};
exports.ClientUpdateOrganizationRequireBehaviorEnum = {
    no_prompt: 'no_prompt',
    pre_login_prompt: 'pre_login_prompt',
    post_login_prompt: 'post_login_prompt',
};
exports.ClientUpdateComplianceLevelEnum = {
    none: 'none',
    fapi1_adv_pkj_par: 'fapi1_adv_pkj_par',
    fapi1_adv_mtls_par: 'fapi1_adv_mtls_par',
    null: 'null',
};
exports.ClientUpdateJwtConfigurationAlgEnum = {
    HS256: 'HS256',
    RS256: 'RS256',
    PS256: 'PS256',
};
exports.ClientUpdateOidcLogoutBackchannelLogoutInitiatorsModeEnum = {
    custom: 'custom',
    all: 'all',
};
exports.ClientUpdateOidcLogoutBackchannelLogoutInitiatorsSelectedInitiatorsEnum = {
    rp_logout: 'rp-logout',
    idp_logout: 'idp-logout',
    password_changed: 'password-changed',
    session_expired: 'session-expired',
    session_revoked: 'session-revoked',
    account_deleted: 'account-deleted',
    email_identifier_changed: 'email-identifier-changed',
    mfa_phone_unenrolled: 'mfa-phone-unenrolled',
    account_deactivated: 'account-deactivated',
};
exports.ConnectionCreateStrategyEnum = {
    ad: 'ad',
    adfs: 'adfs',
    amazon: 'amazon',
    apple: 'apple',
    dropbox: 'dropbox',
    bitbucket: 'bitbucket',
    aol: 'aol',
    auth0_oidc: 'auth0-oidc',
    auth0: 'auth0',
    baidu: 'baidu',
    bitly: 'bitly',
    box: 'box',
    custom: 'custom',
    daccount: 'daccount',
    dwolla: 'dwolla',
    email: 'email',
    evernote_sandbox: 'evernote-sandbox',
    evernote: 'evernote',
    exact: 'exact',
    facebook: 'facebook',
    fitbit: 'fitbit',
    flickr: 'flickr',
    github: 'github',
    google_apps: 'google-apps',
    google_oauth2: 'google-oauth2',
    instagram: 'instagram',
    ip: 'ip',
    line: 'line',
    linkedin: 'linkedin',
    miicard: 'miicard',
    oauth1: 'oauth1',
    oauth2: 'oauth2',
    office365: 'office365',
    oidc: 'oidc',
    okta: 'okta',
    paypal: 'paypal',
    paypal_sandbox: 'paypal-sandbox',
    pingfederate: 'pingfederate',
    planningcenter: 'planningcenter',
    renren: 'renren',
    salesforce_community: 'salesforce-community',
    salesforce_sandbox: 'salesforce-sandbox',
    salesforce: 'salesforce',
    samlp: 'samlp',
    sharepoint: 'sharepoint',
    shopify: 'shopify',
    sms: 'sms',
    soundcloud: 'soundcloud',
    thecity_sandbox: 'thecity-sandbox',
    thecity: 'thecity',
    thirtysevensignals: 'thirtysevensignals',
    twitter: 'twitter',
    untappd: 'untappd',
    vkontakte: 'vkontakte',
    waad: 'waad',
    weibo: 'weibo',
    windowslive: 'windowslive',
    wordpress: 'wordpress',
    yahoo: 'yahoo',
    yammer: 'yammer',
    yandex: 'yandex',
};
exports.ConnectionCreateOptionsPasswordPolicyEnum = {
    none: 'none',
    low: 'low',
    fair: 'fair',
    good: 'good',
    excellent: 'excellent',
    null: 'null',
};
exports.ConnectionCreateOptionsSetUserRootAttributesEnum = {
    each_login: 'on_each_login',
    first_login: 'on_first_login',
};
exports.ConnectionCreateOptionsPasskeyOptionsChallengeUiEnum = {
    both: 'both',
    autofill: 'autofill',
    button: 'button',
};
exports.ConnectionUpdateOptionsPasswordPolicyEnum = {
    none: 'none',
    low: 'low',
    fair: 'fair',
    good: 'good',
    excellent: 'excellent',
    null: 'null',
};
exports.ConnectionUpdateOptionsSetUserRootAttributesEnum = {
    each_login: 'on_each_login',
    first_login: 'on_first_login',
};
exports.CustomDomainStatusEnum = {
    disabled: 'disabled',
    pending: 'pending',
    pending_verification: 'pending_verification',
    ready: 'ready',
};
exports.CustomDomainTypeEnum = {
    auth0_managed_certs: 'auth0_managed_certs',
    self_managed_certs: 'self_managed_certs',
};
exports.DeviceCredentialTypeEnum = {
    public_key: 'public_key',
    refresh_token: 'refresh_token',
    rotating_refresh_token: 'rotating_refresh_token',
};
exports.DeviceCredentialCreateTypeEnum = {
    public_key: 'public_key',
};
exports.EmailProviderCreateNameEnum = {
    mailgun: 'mailgun',
    mandrill: 'mandrill',
    sendgrid: 'sendgrid',
    ses: 'ses',
    sparkpost: 'sparkpost',
    smtp: 'smtp',
    azure_cs: 'azure_cs',
    ms365: 'ms365',
};
exports.EmailProviderUpdateNameEnum = {
    mailgun: 'mailgun',
    mandrill: 'mandrill',
    sendgrid: 'sendgrid',
    ses: 'ses',
    sparkpost: 'sparkpost',
    smtp: 'smtp',
    azure_cs: 'azure_cs',
    ms365: 'ms365',
};
exports.EmailTemplateUpdateTemplateEnum = {
    verify_email: 'verify_email',
    verify_email_by_code: 'verify_email_by_code',
    reset_email: 'reset_email',
    welcome_email: 'welcome_email',
    blocked_account: 'blocked_account',
    stolen_credentials: 'stolen_credentials',
    enrollment_email: 'enrollment_email',
    mfa_oob_code: 'mfa_oob_code',
    user_invitation: 'user_invitation',
    change_password: 'change_password',
    password_reset: 'password_reset',
};
exports.EnrollmentStatusEnum = {
    pending: 'pending',
    confirmed: 'confirmed',
};
exports.FactorNameEnum = {
    push_notification: 'push-notification',
    sms: 'sms',
    email: 'email',
    duo: 'duo',
    otp: 'otp',
    webauthn_roaming: 'webauthn-roaming',
    webauthn_platform: 'webauthn-platform',
    recovery_code: 'recovery-code',
};
exports.GetActionVersions200ResponseVersionsInnerStatusEnum = {
    pending: 'pending',
    building: 'building',
    packaged: 'packaged',
    built: 'built',
    retrying: 'retrying',
    failed: 'failed',
};
exports.GetActions200ResponseActionsInnerStatusEnum = {
    pending: 'pending',
    building: 'building',
    packaged: 'packaged',
    built: 'built',
    retrying: 'retrying',
    failed: 'failed',
};
exports.GetActions200ResponseActionsInnerIntegrationFeatureTypeEnum = {
    unspecified: 'unspecified',
    action: 'action',
    social_connection: 'social_connection',
    log_stream: 'log_stream',
    sso_integration: 'sso_integration',
    sms_provider: 'sms_provider',
};
exports.GetActions200ResponseActionsInnerIntegrationCurrentReleaseRequiredSecretsInnerTypeEnum = {
    UNSPECIFIED: 'UNSPECIFIED',
    STRING: 'STRING',
};
/**
 *
 */
exports.GetActions200ResponseActionsInnerSupportedTriggersInnerIdAnyOf = {
    post_login: 'post-login',
    credentials_exchange: 'credentials-exchange',
    pre_user_registration: 'pre-user-registration',
    post_user_registration: 'post-user-registration',
    post_change_password: 'post-change-password',
    send_phone_message: 'send-phone-message',
    iga_approval: 'iga-approval',
    iga_certification: 'iga-certification',
    iga_fulfillment_assignment: 'iga-fulfillment-assignment',
    iga_fulfillment_execution: 'iga-fulfillment-execution',
    password_reset_post_challenge: 'password-reset-post-challenge',
};
exports.GetAuthenticationMethods200ResponseOneOfInnerTypeEnum = {
    recovery_code: 'recovery-code',
    totp: 'totp',
    push: 'push',
    phone: 'phone',
    email: 'email',
    email_verification: 'email-verification',
    webauthn_roaming: 'webauthn-roaming',
    webauthn_platform: 'webauthn-platform',
    guardian: 'guardian',
    passkey: 'passkey',
};
exports.GetAuthenticationMethods200ResponseOneOfInnerPreferredAuthenticationMethodEnum = {
    sms: 'sms',
    voice: 'voice',
};
exports.GetAuthenticationMethods200ResponseOneOfInnerAuthenticationMethodsInnerTypeEnum = {
    totp: 'totp',
    push: 'push',
    sms: 'sms',
    voice: 'voice',
};
exports.GetBreachedPasswordDetection200ResponseShieldsEnum = {
    block: 'block',
    user_notification: 'user_notification',
    admin_notification: 'admin_notification',
};
exports.GetBreachedPasswordDetection200ResponseAdminNotificationFrequencyEnum = {
    immediately: 'immediately',
    daily: 'daily',
    weekly: 'weekly',
    monthly: 'monthly',
};
exports.GetBreachedPasswordDetection200ResponseMethodEnum = {
    standard: 'standard',
    enhanced: 'enhanced',
};
exports.GetBreachedPasswordDetection200ResponseStagePreUserRegistrationShieldsEnum = {
    block: 'block',
    admin_notification: 'admin_notification',
};
exports.GetBruteForceProtection200ResponseShieldsEnum = {
    block: 'block',
    user_notification: 'user_notification',
};
exports.GetBruteForceProtection200ResponseModeEnum = {
    identifier_and_ip: 'count_per_identifier_and_ip',
    identifier: 'count_per_identifier',
};
exports.GetCredentials200ResponseInnerAlgEnum = {
    RS256: 'RS256',
    RS384: 'RS384',
    PS256: 'PS256',
};
exports.GetEmailTemplatesByTemplateName200ResponseTemplateEnum = {
    verify_email: 'verify_email',
    verify_email_by_code: 'verify_email_by_code',
    reset_email: 'reset_email',
    welcome_email: 'welcome_email',
    blocked_account: 'blocked_account',
    stolen_credentials: 'stolen_credentials',
    enrollment_email: 'enrollment_email',
    mfa_oob_code: 'mfa_oob_code',
    user_invitation: 'user_invitation',
    change_password: 'change_password',
    password_reset: 'password_reset',
};
exports.GetExecution200ResponseStatusEnum = {
    unspecified: 'unspecified',
    pending: 'pending',
    final: 'final',
    partial: 'partial',
    canceled: 'canceled',
    suspended: 'suspended',
};
exports.GetLogStreams200ResponseInnerOneOfStatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
exports.GetLogStreams200ResponseInnerOneOfTypeEnum = {
    http: 'http',
};
exports.GetLogStreams200ResponseInnerOneOf1StatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
exports.GetLogStreams200ResponseInnerOneOf1TypeEnum = {
    eventbridge: 'eventbridge',
};
exports.GetLogStreams200ResponseInnerOneOf1SinkAwsRegionEnum = {
    ap_east_1: 'ap-east-1',
    ap_northeast_1: 'ap-northeast-1',
    ap_northeast_2: 'ap-northeast-2',
    ap_northeast_3: 'ap-northeast-3',
    ap_south_1: 'ap-south-1',
    ap_southeast_1: 'ap-southeast-1',
    ap_southeast_2: 'ap-southeast-2',
    ca_central_1: 'ca-central-1',
    cn_north_1: 'cn-north-1',
    cn_northwest_1: 'cn-northwest-1',
    eu_central_1: 'eu-central-1',
    eu_north_1: 'eu-north-1',
    eu_west_1: 'eu-west-1',
    eu_west_2: 'eu-west-2',
    eu_west_3: 'eu-west-3',
    me_south_1: 'me-south-1',
    sa_east_1: 'sa-east-1',
    us_gov_east_1: 'us-gov-east-1',
    us_gov_west_1: 'us-gov-west-1',
    us_east_1: 'us-east-1',
    us_east_2: 'us-east-2',
    us_west_1: 'us-west-1',
    us_west_2: 'us-west-2',
};
exports.GetLogStreams200ResponseInnerOneOf2StatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
exports.GetLogStreams200ResponseInnerOneOf2TypeEnum = {
    eventgrid: 'eventgrid',
};
exports.GetLogStreams200ResponseInnerOneOf2SinkAzureRegionEnum = {
    australiacentral: 'australiacentral',
    australiaeast: 'australiaeast',
    australiasoutheast: 'australiasoutheast',
    brazilsouth: 'brazilsouth',
    canadacentral: 'canadacentral',
    canadaeast: 'canadaeast',
    centralindia: 'centralindia',
    centralus: 'centralus',
    eastasia: 'eastasia',
    eastus: 'eastus',
    eastus2: 'eastus2',
    francecentral: 'francecentral',
    germanywestcentral: 'germanywestcentral',
    japaneast: 'japaneast',
    japanwest: 'japanwest',
    koreacentral: 'koreacentral',
    koreasouth: 'koreasouth',
    northcentralus: 'northcentralus',
    northeurope: 'northeurope',
    norwayeast: 'norwayeast',
    southafricanorth: 'southafricanorth',
    southcentralus: 'southcentralus',
    southeastasia: 'southeastasia',
    southindia: 'southindia',
    switzerlandnorth: 'switzerlandnorth',
    uaenorth: 'uaenorth',
    uksouth: 'uksouth',
    ukwest: 'ukwest',
    westcentralus: 'westcentralus',
    westeurope: 'westeurope',
    westindia: 'westindia',
    westus: 'westus',
    westus2: 'westus2',
};
exports.GetLogStreams200ResponseInnerOneOf3StatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
exports.GetLogStreams200ResponseInnerOneOf3TypeEnum = {
    datadog: 'datadog',
};
exports.GetLogStreams200ResponseInnerOneOf3SinkDatadogRegionEnum = {
    us: 'us',
    eu: 'eu',
    us3: 'us3',
    us5: 'us5',
};
exports.GetLogStreams200ResponseInnerOneOf4StatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
exports.GetLogStreams200ResponseInnerOneOf4TypeEnum = {
    splunk: 'splunk',
};
exports.GetLogStreams200ResponseInnerOneOf5StatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
exports.GetLogStreams200ResponseInnerOneOf5TypeEnum = {
    sumo: 'sumo',
};
exports.GetLogStreams200ResponseInnerOneOf6StatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
exports.GetLogStreams200ResponseInnerOneOf6TypeEnum = {
    segment: 'segment',
};
exports.GetLogStreams200ResponseInnerOneOf7StatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
exports.GetLogStreams200ResponseInnerOneOf7TypeEnum = {
    mixpanel: 'mixpanel',
};
exports.GetLogStreams200ResponseInnerOneOf7SinkMixpanelRegionEnum = {
    us: 'us',
    eu: 'eu',
};
exports.GetLogStreams200ResponseInnerOneOfFiltersInnerTypeEnum = {
    category: 'category',
};
exports.GetLogStreams200ResponseInnerOneOfFiltersInnerNameEnum = {
    auth_ancillary_fail: 'auth.ancillary.fail',
    auth_ancillary_success: 'auth.ancillary.success',
    auth_login_fail: 'auth.login.fail',
    auth_login_notification: 'auth.login.notification',
    auth_login_success: 'auth.login.success',
    auth_logout_fail: 'auth.logout.fail',
    auth_logout_success: 'auth.logout.success',
    auth_signup_fail: 'auth.signup.fail',
    auth_signup_success: 'auth.signup.success',
    auth_silent_auth_fail: 'auth.silent_auth.fail',
    auth_silent_auth_success: 'auth.silent_auth.success',
    auth_token_exchange_fail: 'auth.token_exchange.fail',
    auth_token_exchange_success: 'auth.token_exchange.success',
    management_fail: 'management.fail',
    management_success: 'management.success',
    system_notification: 'system.notification',
    user_fail: 'user.fail',
    user_notification: 'user.notification',
    user_success: 'user.success',
    other: 'other',
};
exports.GetLogStreams200ResponseInnerOneOfSinkHttpContentFormatEnum = {
    JSONARRAY: 'JSONARRAY',
    JSONLINES: 'JSONLINES',
    JSONOBJECT: 'JSONOBJECT',
};
exports.GetMessageTypes200ResponseMessageTypesEnum = {
    sms: 'sms',
    voice: 'voice',
};
exports.GetPhoneProviders200ResponseProviderEnum = {
    auth0: 'auth0',
    twilio: 'twilio',
    phone_message_hook: 'phone-message-hook',
};
exports.GetPnProviders200ResponseProviderEnum = {
    guardian: 'guardian',
    sns: 'sns',
    direct: 'direct',
};
exports.GetSuspiciousIpThrottling200ResponseShieldsEnum = {
    block: 'block',
    admin_notification: 'admin_notification',
};
exports.HookCreateTriggerIdEnum = {
    credentials_exchange: 'credentials-exchange',
    pre_user_registration: 'pre-user-registration',
    post_user_registration: 'post-user-registration',
    post_change_password: 'post-change-password',
    send_phone_message: 'send-phone-message',
};
exports.JobFormatEnum = {
    json: 'json',
    csv: 'csv',
};
exports.PatchAuthenticationMethodsByAuthenticationMethodIdRequestPreferredAuthenticationMethodEnum = {
    voice: 'voice',
    sms: 'sms',
};
exports.PatchBindingsRequestBindingsInnerOneOfRefTypeEnum = {
    binding_id: 'binding_id',
    action_id: 'action_id',
    action_name: 'action_name',
};
exports.PatchBreachedPasswordDetectionRequestShieldsEnum = {
    block: 'block',
    user_notification: 'user_notification',
    admin_notification: 'admin_notification',
};
exports.PatchBreachedPasswordDetectionRequestAdminNotificationFrequencyEnum = {
    immediately: 'immediately',
    daily: 'daily',
    weekly: 'weekly',
    monthly: 'monthly',
};
exports.PatchBreachedPasswordDetectionRequestMethodEnum = {
    standard: 'standard',
    enhanced: 'enhanced',
};
exports.PatchBreachedPasswordDetectionRequestStagePreUserRegistrationShieldsEnum = {
    block: 'block',
    admin_notification: 'admin_notification',
};
exports.PatchBruteForceProtectionRequestShieldsEnum = {
    block: 'block',
    user_notification: 'user_notification',
};
exports.PatchBruteForceProtectionRequestModeEnum = {
    identifier_and_ip: 'count_per_identifier_and_ip',
    identifier: 'count_per_identifier',
};
exports.PatchClientGrantsByIdRequestOrganizationUsageEnum = {
    deny: 'deny',
    allow: 'allow',
    require: 'require',
    null: 'null',
};
exports.PatchCustomDomainsByIdRequestTlsPolicyEnum = {
    recommended: 'recommended',
    compatible: 'compatible',
};
exports.PatchCustomDomainsByIdRequestCustomClientIpHeaderEnum = {
    true_client_ip: 'true-client-ip',
    cf_connecting_ip: 'cf-connecting-ip',
    x_forwarded_for: 'x-forwarded-for',
    x_azure_clientip: 'x-azure-clientip',
    empty: '',
};
exports.PatchEmailTemplatesByTemplateNameRequestTemplateEnum = {
    verify_email: 'verify_email',
    verify_email_by_code: 'verify_email_by_code',
    reset_email: 'reset_email',
    welcome_email: 'welcome_email',
    blocked_account: 'blocked_account',
    stolen_credentials: 'stolen_credentials',
    enrollment_email: 'enrollment_email',
    mfa_oob_code: 'mfa_oob_code',
    user_invitation: 'user_invitation',
    change_password: 'change_password',
    password_reset: 'password_reset',
};
exports.PatchLogStreamsByIdRequestStatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
exports.PatchLogStreamsByIdRequestSinkOneOfDatadogRegionEnum = {
    us: 'us',
    eu: 'eu',
    us3: 'us3',
    us5: 'us5',
};
exports.PatchLogStreamsByIdRequestSinkOneOf3MixpanelRegionEnum = {
    us: 'us',
    eu: 'eu',
};
exports.PatchSuspiciousIpThrottlingRequestShieldsEnum = {
    block: 'block',
    admin_notification: 'admin_notification',
};
exports.PostAuthenticationMethods201ResponseTypeEnum = {
    phone: 'phone',
    email: 'email',
    totp: 'totp',
    webauthn_roaming: 'webauthn-roaming',
};
exports.PostAuthenticationMethods201ResponsePreferredAuthenticationMethodEnum = {
    voice: 'voice',
    sms: 'sms',
};
exports.PostAuthenticationMethodsRequestTypeEnum = {
    phone: 'phone',
    email: 'email',
    totp: 'totp',
    webauthn_roaming: 'webauthn-roaming',
};
exports.PostAuthenticationMethodsRequestPreferredAuthenticationMethodEnum = {
    voice: 'voice',
    sms: 'sms',
};
exports.PostBrandingThemeRequestBordersButtonsStyleEnum = {
    pill: 'pill',
    rounded: 'rounded',
    sharp: 'sharp',
};
exports.PostBrandingThemeRequestBordersInputsStyleEnum = {
    pill: 'pill',
    rounded: 'rounded',
    sharp: 'sharp',
};
exports.PostBrandingThemeRequestFontsLinksStyleEnum = {
    normal: 'normal',
    underlined: 'underlined',
};
exports.PostBrandingThemeRequestPageBackgroundPageLayoutEnum = {
    center: 'center',
    left: 'left',
    right: 'right',
};
exports.PostBrandingThemeRequestWidgetHeaderTextAlignmentEnum = {
    center: 'center',
    left: 'left',
    right: 'right',
};
exports.PostBrandingThemeRequestWidgetLogoPositionEnum = {
    center: 'center',
    left: 'left',
    none: 'none',
    right: 'right',
};
exports.PostBrandingThemeRequestWidgetSocialButtonsLayoutEnum = {
    bottom: 'bottom',
    top: 'top',
};
exports.PostCredentialsRequestCredentialTypeEnum = {
    public_key: 'public_key',
    cert_subject_dn: 'cert_subject_dn',
    x509_cert: 'x509_cert',
};
exports.PostCustomDomains201ResponseStatusEnum = {
    disabled: 'disabled',
    pending: 'pending',
    pending_verification: 'pending_verification',
    ready: 'ready',
};
exports.PostCustomDomains201ResponseTypeEnum = {
    auth0_managed_certs: 'auth0_managed_certs',
    self_managed_certs: 'self_managed_certs',
};
exports.PostCustomDomains201ResponseVerificationMethodsInnerNameEnum = {
    cname: 'cname',
    txt: 'txt',
};
exports.PostCustomDomainsRequestTypeEnum = {
    auth0_managed_certs: 'auth0_managed_certs',
    self_managed_certs: 'self_managed_certs',
};
exports.PostCustomDomainsRequestVerificationMethodEnum = {
    txt: 'txt',
};
exports.PostCustomDomainsRequestTlsPolicyEnum = {
    recommended: 'recommended',
    compatible: 'compatible',
};
exports.PostCustomDomainsRequestCustomClientIpHeaderEnum = {
    true_client_ip: 'true-client-ip',
    cf_connecting_ip: 'cf-connecting-ip',
    x_forwarded_for: 'x-forwarded-for',
    x_azure_clientip: 'x-azure-clientip',
    null: 'null',
};
exports.PostEmailTemplatesRequestTemplateEnum = {
    verify_email: 'verify_email',
    verify_email_by_code: 'verify_email_by_code',
    reset_email: 'reset_email',
    welcome_email: 'welcome_email',
    blocked_account: 'blocked_account',
    stolen_credentials: 'stolen_credentials',
    enrollment_email: 'enrollment_email',
    mfa_oob_code: 'mfa_oob_code',
    user_invitation: 'user_invitation',
    change_password: 'change_password',
    password_reset: 'password_reset',
};
exports.PostFlowsVaultConnectionsRequestAnyOfAppIdEnum = {
    ACTIVECAMPAIGN: 'ACTIVECAMPAIGN',
};
exports.PostFlowsVaultConnectionsRequestAnyOf1AppIdEnum = {
    AIRTABLE: 'AIRTABLE',
};
exports.PostFlowsVaultConnectionsRequestAnyOf10AppIdEnum = {
    MAILCHIMP: 'MAILCHIMP',
};
exports.PostFlowsVaultConnectionsRequestAnyOf11AppIdEnum = {
    MAILJET: 'MAILJET',
};
exports.PostFlowsVaultConnectionsRequestAnyOf11SetupTypeEnum = {
    API_KEY: 'API_KEY',
};
exports.PostFlowsVaultConnectionsRequestAnyOf12AppIdEnum = {
    PIPEDRIVE: 'PIPEDRIVE',
};
exports.PostFlowsVaultConnectionsRequestAnyOf12SetupAnyOfTypeEnum = {
    TOKEN: 'TOKEN',
};
exports.PostFlowsVaultConnectionsRequestAnyOf13AppIdEnum = {
    SALESFORCE: 'SALESFORCE',
};
exports.PostFlowsVaultConnectionsRequestAnyOf14AppIdEnum = {
    SENDGRID: 'SENDGRID',
};
exports.PostFlowsVaultConnectionsRequestAnyOf15AppIdEnum = {
    SLACK: 'SLACK',
};
exports.PostFlowsVaultConnectionsRequestAnyOf15SetupAnyOfTypeEnum = {
    WEBHOOK: 'WEBHOOK',
};
exports.PostFlowsVaultConnectionsRequestAnyOf16AppIdEnum = {
    STRIPE: 'STRIPE',
};
exports.PostFlowsVaultConnectionsRequestAnyOf16SetupAnyOfTypeEnum = {
    KEY_PAIR: 'KEY_PAIR',
};
exports.PostFlowsVaultConnectionsRequestAnyOf16SetupAnyOf1TypeEnum = {
    OAUTH_CODE: 'OAUTH_CODE',
};
exports.PostFlowsVaultConnectionsRequestAnyOf16SetupAnyOf1EnvironmentEnum = {
    live: 'live',
    test: 'test',
};
exports.PostFlowsVaultConnectionsRequestAnyOf17AppIdEnum = {
    TELEGRAM: 'TELEGRAM',
};
exports.PostFlowsVaultConnectionsRequestAnyOf18AppIdEnum = {
    TWILIO: 'TWILIO',
};
exports.PostFlowsVaultConnectionsRequestAnyOf18SetupTypeEnum = {
    API_KEY: 'API_KEY',
};
exports.PostFlowsVaultConnectionsRequestAnyOf19AppIdEnum = {
    WHATSAPP: 'WHATSAPP',
};
exports.PostFlowsVaultConnectionsRequestAnyOf1SetupTypeEnum = {
    API_KEY: 'API_KEY',
};
exports.PostFlowsVaultConnectionsRequestAnyOf2AppIdEnum = {
    AUTH0: 'AUTH0',
};
exports.PostFlowsVaultConnectionsRequestAnyOf20AppIdEnum = {
    ZAPIER: 'ZAPIER',
};
exports.PostFlowsVaultConnectionsRequestAnyOf2SetupTypeEnum = {
    OAUTH_APP: 'OAUTH_APP',
};
exports.PostFlowsVaultConnectionsRequestAnyOf3AppIdEnum = {
    BIGQUERY: 'BIGQUERY',
};
exports.PostFlowsVaultConnectionsRequestAnyOf3SetupTypeEnum = {
    OAUTH_JWT: 'OAUTH_JWT',
};
exports.PostFlowsVaultConnectionsRequestAnyOf4AppIdEnum = {
    CLEARBIT: 'CLEARBIT',
};
exports.PostFlowsVaultConnectionsRequestAnyOf4SetupTypeEnum = {
    API_KEY: 'API_KEY',
};
exports.PostFlowsVaultConnectionsRequestAnyOf5AppIdEnum = {
    DOCUSIGN: 'DOCUSIGN',
};
exports.PostFlowsVaultConnectionsRequestAnyOf5SetupTypeEnum = {
    OAUTH_CODE: 'OAUTH_CODE',
};
exports.PostFlowsVaultConnectionsRequestAnyOf6AppIdEnum = {
    GOOGLE_SHEETS: 'GOOGLE_SHEETS',
};
exports.PostFlowsVaultConnectionsRequestAnyOf7AppIdEnum = {
    HTTP: 'HTTP',
};
exports.PostFlowsVaultConnectionsRequestAnyOf7SetupTypeEnum = {
    BEARER: 'BEARER',
};
exports.PostFlowsVaultConnectionsRequestAnyOf8AppIdEnum = {
    HUBSPOT: 'HUBSPOT',
};
exports.PostFlowsVaultConnectionsRequestAnyOf9AppIdEnum = {
    JWT: 'JWT',
};
exports.PostFlowsVaultConnectionsRequestAnyOf9SetupTypeEnum = {
    JWT: 'JWT',
};
exports.PostFlowsVaultConnectionsRequestAnyOf9SetupAlgorithmEnum = {
    HS256: 'HS256',
    HS384: 'HS384',
    HS512: 'HS512',
    RS256: 'RS256',
    RS384: 'RS384',
    RS512: 'RS512',
    ES256: 'ES256',
    ES384: 'ES384',
    ES512: 'ES512',
    PS256: 'PS256',
    PS384: 'PS384',
    PS512: 'PS512',
};
exports.PostFlowsVaultConnectionsRequestAnyOfSetupTypeEnum = {
    API_KEY: 'API_KEY',
};
exports.PostFormsRequestEndingResumeFlowEnum = {
    true: true,
};
exports.PostFormsRequestNodesInnerAnyOfTypeEnum = {
    FLOW: 'FLOW',
};
exports.PostFormsRequestNodesInnerAnyOf1TypeEnum = {
    ROUTER: 'ROUTER',
};
exports.PostFormsRequestNodesInnerAnyOf2TypeEnum = {
    STEP: 'STEP',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf1AnyOfTypeEnum = {
    GMAPS_ADDRESS: 'GMAPS_ADDRESS',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf1AnyOf1TypeEnum = {
    RECAPTCHA: 'RECAPTCHA',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOfTypeEnum = {
    BOOLEAN: 'BOOLEAN',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf1TypeEnum = {
    CARDS: 'CARDS',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf10TypeEnum = {
    PASSWORD: 'PASSWORD',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf10ConfigHashEnum = {
    NONE: 'NONE',
    MD5: 'MD5',
    SHA1: 'SHA1',
    SHA256: 'SHA256',
    SHA512: 'SHA512',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11TypeEnum = {
    PAYMENT: 'PAYMENT',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11ConfigProviderEnum = {
    STRIPE: 'STRIPE',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11ConfigChargeAnyOfTypeEnum = {
    ONE_OFF: 'ONE_OFF',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11ConfigChargeAnyOf1TypeEnum = {
    SUBSCRIPTION: 'SUBSCRIPTION',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11ConfigChargeAnyOfOneOffCurrencyEnum = {
    AUD: 'AUD',
    CAD: 'CAD',
    CHF: 'CHF',
    EUR: 'EUR',
    GBP: 'GBP',
    INR: 'INR',
    MXN: 'MXN',
    SEK: 'SEK',
    USD: 'USD',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf12TypeEnum = {
    SOCIAL: 'SOCIAL',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf13TypeEnum = {
    TEL: 'TEL',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf14TypeEnum = {
    TEXT: 'TEXT',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf15TypeEnum = {
    URL: 'URL',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf2TypeEnum = {
    CHOICE: 'CHOICE',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf3TypeEnum = {
    CUSTOM: 'CUSTOM',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf4TypeEnum = {
    DATE: 'DATE',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf5TypeEnum = {
    DROPDOWN: 'DROPDOWN',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf6TypeEnum = {
    EMAIL: 'EMAIL',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf7TypeEnum = {
    FILE: 'FILE',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf7ConfigCategoriesEnum = {
    AUDIO: 'AUDIO',
    VIDEO: 'VIDEO',
    IMAGE: 'IMAGE',
    DOCUMENT: 'DOCUMENT',
    ARCHIVE: 'ARCHIVE',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf8TypeEnum = {
    LEGAL: 'LEGAL',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf9TypeEnum = {
    NUMBER: 'NUMBER',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOfTypeEnum = {
    DIVIDER: 'DIVIDER',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf1TypeEnum = {
    HTML: 'HTML',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf2TypeEnum = {
    IMAGE: 'IMAGE',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf2ConfigPositionEnum = {
    LEFT: 'LEFT',
    CENTER: 'CENTER',
    RIGHT: 'RIGHT',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf3TypeEnum = {
    JUMP_BUTTON: 'JUMP_BUTTON',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf4TypeEnum = {
    NEXT_BUTTON: 'NEXT_BUTTON',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf5TypeEnum = {
    PREVIOUS_BUTTON: 'PREVIOUS_BUTTON',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf6TypeEnum = {
    RICH_TEXT: 'RICH_TEXT',
};
/**
 *
 */
exports.PostFormsRequestNodesInnerAnyOfConfigNextNodeAnyOf = {
    ending: '$ending',
};
exports.PostIdentitiesRequestProviderEnum = {
    ad: 'ad',
    adfs: 'adfs',
    amazon: 'amazon',
    apple: 'apple',
    dropbox: 'dropbox',
    bitbucket: 'bitbucket',
    aol: 'aol',
    auth0_oidc: 'auth0-oidc',
    auth0: 'auth0',
    baidu: 'baidu',
    bitly: 'bitly',
    box: 'box',
    custom: 'custom',
    daccount: 'daccount',
    dwolla: 'dwolla',
    email: 'email',
    evernote_sandbox: 'evernote-sandbox',
    evernote: 'evernote',
    exact: 'exact',
    facebook: 'facebook',
    fitbit: 'fitbit',
    flickr: 'flickr',
    github: 'github',
    google_apps: 'google-apps',
    google_oauth2: 'google-oauth2',
    instagram: 'instagram',
    ip: 'ip',
    line: 'line',
    linkedin: 'linkedin',
    miicard: 'miicard',
    oauth1: 'oauth1',
    oauth2: 'oauth2',
    office365: 'office365',
    oidc: 'oidc',
    okta: 'okta',
    paypal: 'paypal',
    paypal_sandbox: 'paypal-sandbox',
    pingfederate: 'pingfederate',
    planningcenter: 'planningcenter',
    renren: 'renren',
    salesforce_community: 'salesforce-community',
    salesforce_sandbox: 'salesforce-sandbox',
    salesforce: 'salesforce',
    samlp: 'samlp',
    sharepoint: 'sharepoint',
    shopify: 'shopify',
    sms: 'sms',
    soundcloud: 'soundcloud',
    thecity_sandbox: 'thecity-sandbox',
    thecity: 'thecity',
    thirtysevensignals: 'thirtysevensignals',
    twitter: 'twitter',
    untappd: 'untappd',
    vkontakte: 'vkontakte',
    waad: 'waad',
    weibo: 'weibo',
    windowslive: 'windowslive',
    wordpress: 'wordpress',
    yahoo: 'yahoo',
    yammer: 'yammer',
    yandex: 'yandex',
};
exports.PostLogStreamsRequestOneOfTypeEnum = {
    http: 'http',
};
exports.PostLogStreamsRequestOneOf1TypeEnum = {
    eventbridge: 'eventbridge',
};
exports.PostLogStreamsRequestOneOf1SinkAwsRegionEnum = {
    ap_east_1: 'ap-east-1',
    ap_northeast_1: 'ap-northeast-1',
    ap_northeast_2: 'ap-northeast-2',
    ap_northeast_3: 'ap-northeast-3',
    ap_south_1: 'ap-south-1',
    ap_southeast_1: 'ap-southeast-1',
    ap_southeast_2: 'ap-southeast-2',
    ca_central_1: 'ca-central-1',
    cn_north_1: 'cn-north-1',
    cn_northwest_1: 'cn-northwest-1',
    eu_central_1: 'eu-central-1',
    eu_north_1: 'eu-north-1',
    eu_west_1: 'eu-west-1',
    eu_west_2: 'eu-west-2',
    eu_west_3: 'eu-west-3',
    me_south_1: 'me-south-1',
    sa_east_1: 'sa-east-1',
    us_gov_east_1: 'us-gov-east-1',
    us_gov_west_1: 'us-gov-west-1',
    us_east_1: 'us-east-1',
    us_east_2: 'us-east-2',
    us_west_1: 'us-west-1',
    us_west_2: 'us-west-2',
};
exports.PostLogStreamsRequestOneOf2TypeEnum = {
    eventgrid: 'eventgrid',
};
exports.PostLogStreamsRequestOneOf2SinkAzureRegionEnum = {
    australiacentral: 'australiacentral',
    australiaeast: 'australiaeast',
    australiasoutheast: 'australiasoutheast',
    brazilsouth: 'brazilsouth',
    canadacentral: 'canadacentral',
    canadaeast: 'canadaeast',
    centralindia: 'centralindia',
    centralus: 'centralus',
    eastasia: 'eastasia',
    eastus: 'eastus',
    eastus2: 'eastus2',
    francecentral: 'francecentral',
    germanywestcentral: 'germanywestcentral',
    japaneast: 'japaneast',
    japanwest: 'japanwest',
    koreacentral: 'koreacentral',
    koreasouth: 'koreasouth',
    northcentralus: 'northcentralus',
    northeurope: 'northeurope',
    norwayeast: 'norwayeast',
    southafricanorth: 'southafricanorth',
    southcentralus: 'southcentralus',
    southeastasia: 'southeastasia',
    southindia: 'southindia',
    switzerlandnorth: 'switzerlandnorth',
    uaenorth: 'uaenorth',
    uksouth: 'uksouth',
    ukwest: 'ukwest',
    westcentralus: 'westcentralus',
    westeurope: 'westeurope',
    westindia: 'westindia',
    westus: 'westus',
    westus2: 'westus2',
};
exports.PostLogStreamsRequestOneOf3TypeEnum = {
    datadog: 'datadog',
};
exports.PostLogStreamsRequestOneOf4TypeEnum = {
    splunk: 'splunk',
};
exports.PostLogStreamsRequestOneOf5TypeEnum = {
    sumo: 'sumo',
};
exports.PostLogStreamsRequestOneOf6TypeEnum = {
    segment: 'segment',
};
exports.PostLogStreamsRequestOneOf7TypeEnum = {
    mixpanel: 'mixpanel',
};
exports.PostLogStreamsRequestOneOfFiltersInnerTypeEnum = {
    category: 'category',
};
exports.PostLogStreamsRequestOneOfFiltersInnerNameEnum = {
    auth_ancillary_fail: 'auth.ancillary.fail',
    auth_ancillary_success: 'auth.ancillary.success',
    auth_login_fail: 'auth.login.fail',
    auth_login_notification: 'auth.login.notification',
    auth_login_success: 'auth.login.success',
    auth_logout_fail: 'auth.logout.fail',
    auth_logout_success: 'auth.logout.success',
    auth_signup_fail: 'auth.signup.fail',
    auth_signup_success: 'auth.signup.success',
    auth_silent_auth_fail: 'auth.silent_auth.fail',
    auth_silent_auth_success: 'auth.silent_auth.success',
    auth_token_exchange_fail: 'auth.token_exchange.fail',
    auth_token_exchange_success: 'auth.token_exchange.success',
    management_fail: 'management.fail',
    management_success: 'management.success',
    system_notification: 'system.notification',
    user_fail: 'user.fail',
    user_notification: 'user.notification',
    user_success: 'user.success',
    other: 'other',
};
exports.PostUsersExportsRequestFormatEnum = {
    json: 'json',
    csv: 'csv',
};
exports.PostVerificationEmailRequestIdentityProviderEnum = {
    ad: 'ad',
    adfs: 'adfs',
    amazon: 'amazon',
    apple: 'apple',
    dropbox: 'dropbox',
    bitbucket: 'bitbucket',
    aol: 'aol',
    auth0_oidc: 'auth0-oidc',
    auth0: 'auth0',
    baidu: 'baidu',
    bitly: 'bitly',
    box: 'box',
    custom: 'custom',
    daccount: 'daccount',
    dwolla: 'dwolla',
    email: 'email',
    evernote_sandbox: 'evernote-sandbox',
    evernote: 'evernote',
    exact: 'exact',
    facebook: 'facebook',
    fitbit: 'fitbit',
    flickr: 'flickr',
    github: 'github',
    google_apps: 'google-apps',
    google_oauth2: 'google-oauth2',
    instagram: 'instagram',
    ip: 'ip',
    line: 'line',
    linkedin: 'linkedin',
    miicard: 'miicard',
    oauth1: 'oauth1',
    oauth2: 'oauth2',
    office365: 'office365',
    oidc: 'oidc',
    okta: 'okta',
    paypal: 'paypal',
    paypal_sandbox: 'paypal-sandbox',
    pingfederate: 'pingfederate',
    planningcenter: 'planningcenter',
    renren: 'renren',
    salesforce_community: 'salesforce-community',
    salesforce_sandbox: 'salesforce-sandbox',
    salesforce: 'salesforce',
    samlp: 'samlp',
    sharepoint: 'sharepoint',
    shopify: 'shopify',
    sms: 'sms',
    soundcloud: 'soundcloud',
    thecity_sandbox: 'thecity-sandbox',
    thecity: 'thecity',
    thirtysevensignals: 'thirtysevensignals',
    twitter: 'twitter',
    untappd: 'untappd',
    vkontakte: 'vkontakte',
    waad: 'waad',
    weibo: 'weibo',
    windowslive: 'windowslive',
    wordpress: 'wordpress',
    yahoo: 'yahoo',
    yammer: 'yammer',
    yandex: 'yandex',
};
exports.PostVerify200ResponseStatusEnum = {
    disabled: 'disabled',
    pending: 'pending',
    pending_verification: 'pending_verification',
    ready: 'ready',
};
exports.PostVerify200ResponseTypeEnum = {
    auth0_managed_certs: 'auth0_managed_certs',
    self_managed_certs: 'self_managed_certs',
};
exports.PromptsSettingsUniversalLoginExperienceEnum = {
    new: 'new',
    classic: 'classic',
};
exports.PromptsSettingsUpdateUniversalLoginExperienceEnum = {
    new: 'new',
    classic: 'classic',
};
exports.PutAuthenticationMethods200ResponseInnerTypeEnum = {
    phone: 'phone',
    email: 'email',
    totp: 'totp',
    webauthn_roaming: 'webauthn-roaming',
};
exports.PutAuthenticationMethods200ResponseInnerPreferredAuthenticationMethodEnum = {
    voice: 'voice',
    sms: 'sms',
};
exports.PutAuthenticationMethods200ResponseInnerAuthenticationMethodsInnerTypeEnum = {
    totp: 'totp',
    push: 'push',
    sms: 'sms',
    voice: 'voice',
};
exports.PutAuthenticationMethodsRequestInnerTypeEnum = {
    phone: 'phone',
    email: 'email',
    totp: 'totp',
};
exports.PutAuthenticationMethodsRequestInnerPreferredAuthenticationMethodEnum = {
    voice: 'voice',
    sms: 'sms',
};
exports.ResourceServerSigningAlgEnum = {
    HS256: 'HS256',
    RS256: 'RS256',
    PS256: 'PS256',
};
exports.ResourceServerTokenDialectEnum = {
    access_token: 'access_token',
    access_token_authz: 'access_token_authz',
    rfc9068_profile: 'rfc9068_profile',
    rfc9068_profile_authz: 'rfc9068_profile_authz',
};
exports.ResourceServerConsentPolicyEnum = {
    transactional_authorization_with_mfa: 'transactional-authorization-with-mfa',
    null: 'null',
};
exports.ResourceServerCreateSigningAlgEnum = {
    HS256: 'HS256',
    RS256: 'RS256',
    PS256: 'PS256',
};
exports.ResourceServerCreateTokenDialectEnum = {
    access_token: 'access_token',
    access_token_authz: 'access_token_authz',
    rfc9068_profile: 'rfc9068_profile',
    rfc9068_profile_authz: 'rfc9068_profile_authz',
};
exports.ResourceServerCreateConsentPolicyEnum = {
    transactional_authorization_with_mfa: 'transactional-authorization-with-mfa',
    null: 'null',
};
exports.ResourceServerProofOfPossessionMechanismEnum = {
    mtls: 'mtls',
};
exports.ResourceServerTokenEncryptionFormatEnum = {
    compact_nested_jwe: 'compact-nested-jwe',
};
exports.ResourceServerTokenEncryptionEncryptionKeyAlgEnum = {
    _256: 'RSA-OAEP-256',
    _384: 'RSA-OAEP-384',
    _512: 'RSA-OAEP-512',
};
exports.ResourceServerUpdateSigningAlgEnum = {
    HS256: 'HS256',
    RS256: 'RS256',
    PS256: 'PS256',
};
exports.ResourceServerUpdateTokenDialectEnum = {
    access_token: 'access_token',
    access_token_authz: 'access_token_authz',
    rfc9068_profile: 'rfc9068_profile',
    rfc9068_profile_authz: 'rfc9068_profile_authz',
};
exports.ResourceServerUpdateConsentPolicyEnum = {
    transactional_authorization_with_mfa: 'transactional-authorization-with-mfa',
    null: 'null',
};
exports.SsProfileAllowedStrategiesEnum = {
    oidc: 'oidc',
    samlp: 'samlp',
    waad: 'waad',
    google_apps: 'google-apps',
    adfs: 'adfs',
    okta: 'okta',
    keycloak_samlp: 'keycloak-samlp',
    pingfederate: 'pingfederate',
};
exports.SsProfileCreateAllowedStrategiesEnum = {
    oidc: 'oidc',
    samlp: 'samlp',
    waad: 'waad',
    google_apps: 'google-apps',
    adfs: 'adfs',
    okta: 'okta',
    keycloak_samlp: 'keycloak-samlp',
    pingfederate: 'pingfederate',
};
exports.SsProfileUpdateAllowedStrategiesEnum = {
    oidc: 'oidc',
    samlp: 'samlp',
    waad: 'waad',
    google_apps: 'google-apps',
    adfs: 'adfs',
    okta: 'okta',
    keycloak_samlp: 'keycloak-samlp',
    pingfederate: 'pingfederate',
};
exports.TenantSettingsEnabledLocalesEnum = {
    ar: 'ar',
    bg: 'bg',
    bs: 'bs',
    ca_ES: 'ca-ES',
    cs: 'cs',
    cy: 'cy',
    da: 'da',
    de: 'de',
    el: 'el',
    en: 'en',
    es: 'es',
    et: 'et',
    eu_ES: 'eu-ES',
    fi: 'fi',
    fr: 'fr',
    fr_CA: 'fr-CA',
    fr_FR: 'fr-FR',
    gl_ES: 'gl-ES',
    he: 'he',
    hi: 'hi',
    hr: 'hr',
    hu: 'hu',
    id: 'id',
    is: 'is',
    it: 'it',
    ja: 'ja',
    ko: 'ko',
    lt: 'lt',
    lv: 'lv',
    nb: 'nb',
    nl: 'nl',
    nn: 'nn',
    no: 'no',
    pl: 'pl',
    pt: 'pt',
    pt_BR: 'pt-BR',
    pt_PT: 'pt-PT',
    ro: 'ro',
    ru: 'ru',
    sk: 'sk',
    sl: 'sl',
    sr: 'sr',
    sv: 'sv',
    th: 'th',
    tr: 'tr',
    uk: 'uk',
    vi: 'vi',
    zh_CN: 'zh-CN',
    zh_TW: 'zh-TW',
};
exports.TenantSettingsDeviceFlowCharsetEnum = {
    base20: 'base20',
    digits: 'digits',
};
exports.TenantSettingsSessionCookieModeEnum = {
    persistent: 'persistent',
    non_persistent: 'non-persistent',
};
exports.TenantSettingsUpdateEnabledLocalesEnum = {
    ar: 'ar',
    bg: 'bg',
    bs: 'bs',
    ca_ES: 'ca-ES',
    cs: 'cs',
    cy: 'cy',
    da: 'da',
    de: 'de',
    el: 'el',
    en: 'en',
    es: 'es',
    et: 'et',
    eu_ES: 'eu-ES',
    fi: 'fi',
    fr: 'fr',
    fr_CA: 'fr-CA',
    fr_FR: 'fr-FR',
    gl_ES: 'gl-ES',
    he: 'he',
    hi: 'hi',
    hr: 'hr',
    hu: 'hu',
    id: 'id',
    is: 'is',
    it: 'it',
    ja: 'ja',
    ko: 'ko',
    lt: 'lt',
    lv: 'lv',
    nb: 'nb',
    nl: 'nl',
    nn: 'nn',
    no: 'no',
    pl: 'pl',
    pt: 'pt',
    pt_BR: 'pt-BR',
    pt_PT: 'pt-PT',
    ro: 'ro',
    ru: 'ru',
    sk: 'sk',
    sl: 'sl',
    sr: 'sr',
    sv: 'sv',
    th: 'th',
    tr: 'tr',
    uk: 'uk',
    vi: 'vi',
    zh_CN: 'zh-CN',
    zh_TW: 'zh-TW',
};
exports.TenantSettingsUpdateDeviceFlowCharsetEnum = {
    base20: 'base20',
    digits: 'digits',
};
exports.TenantSettingsUpdateFlagsChangePwdFlowV1Enum = {
    false: false,
};
exports.UserEnrollmentStatusEnum = {
    pending: 'pending',
    confirmed: 'confirmed',
};
exports.UserEnrollmentAuthMethodEnum = {
    authenticator: 'authenticator',
    guardian: 'guardian',
    sms: 'sms',
    webauthn_platform: 'webauthn-platform',
    webauthn_roaming: 'webauthn-roaming',
};
/**
 *
 */
exports.GetConnectionsStrategyEnum = {
    ad: 'ad',
    adfs: 'adfs',
    amazon: 'amazon',
    apple: 'apple',
    dropbox: 'dropbox',
    bitbucket: 'bitbucket',
    aol: 'aol',
    auth0_oidc: 'auth0-oidc',
    auth0: 'auth0',
    baidu: 'baidu',
    bitly: 'bitly',
    box: 'box',
    custom: 'custom',
    daccount: 'daccount',
    dwolla: 'dwolla',
    email: 'email',
    evernote_sandbox: 'evernote-sandbox',
    evernote: 'evernote',
    exact: 'exact',
    facebook: 'facebook',
    fitbit: 'fitbit',
    flickr: 'flickr',
    github: 'github',
    google_apps: 'google-apps',
    google_oauth2: 'google-oauth2',
    instagram: 'instagram',
    ip: 'ip',
    line: 'line',
    linkedin: 'linkedin',
    miicard: 'miicard',
    oauth1: 'oauth1',
    oauth2: 'oauth2',
    office365: 'office365',
    oidc: 'oidc',
    okta: 'okta',
    paypal: 'paypal',
    paypal_sandbox: 'paypal-sandbox',
    pingfederate: 'pingfederate',
    planningcenter: 'planningcenter',
    renren: 'renren',
    salesforce_community: 'salesforce-community',
    salesforce_sandbox: 'salesforce-sandbox',
    salesforce: 'salesforce',
    samlp: 'samlp',
    sharepoint: 'sharepoint',
    shopify: 'shopify',
    sms: 'sms',
    soundcloud: 'soundcloud',
    thecity_sandbox: 'thecity-sandbox',
    thecity: 'thecity',
    thirtysevensignals: 'thirtysevensignals',
    twitter: 'twitter',
    untappd: 'untappd',
    vkontakte: 'vkontakte',
    waad: 'waad',
    weibo: 'weibo',
    windowslive: 'windowslive',
    wordpress: 'wordpress',
    yahoo: 'yahoo',
    yammer: 'yammer',
    yandex: 'yandex',
    auth0_adldap: 'auth0-adldap',
};
/**
 *
 */
exports.GetDeviceCredentialsTypeEnum = {
    public_key: 'public_key',
    refresh_token: 'refresh_token',
    rotating_refresh_token: 'rotating_refresh_token',
};
/**
 *
 */
exports.GetEmailTemplatesByTemplateNameTemplateNameEnum = {
    verify_email: 'verify_email',
    verify_email_by_code: 'verify_email_by_code',
    reset_email: 'reset_email',
    welcome_email: 'welcome_email',
    blocked_account: 'blocked_account',
    stolen_credentials: 'stolen_credentials',
    enrollment_email: 'enrollment_email',
    mfa_oob_code: 'mfa_oob_code',
    user_invitation: 'user_invitation',
    change_password: 'change_password',
    password_reset: 'password_reset',
};
/**
 *
 */
exports.PatchEmailTemplatesByTemplateNameOperationTemplateNameEnum = {
    verify_email: 'verify_email',
    verify_email_by_code: 'verify_email_by_code',
    reset_email: 'reset_email',
    welcome_email: 'welcome_email',
    blocked_account: 'blocked_account',
    stolen_credentials: 'stolen_credentials',
    enrollment_email: 'enrollment_email',
    mfa_oob_code: 'mfa_oob_code',
    user_invitation: 'user_invitation',
    change_password: 'change_password',
    password_reset: 'password_reset',
};
/**
 *
 */
exports.PutEmailTemplatesByTemplateNameTemplateNameEnum = {
    verify_email: 'verify_email',
    verify_email_by_code: 'verify_email_by_code',
    reset_email: 'reset_email',
    welcome_email: 'welcome_email',
    blocked_account: 'blocked_account',
    stolen_credentials: 'stolen_credentials',
    enrollment_email: 'enrollment_email',
    mfa_oob_code: 'mfa_oob_code',
    user_invitation: 'user_invitation',
    change_password: 'change_password',
    password_reset: 'password_reset',
};
/**
 *
 */
exports.GetFlowsHydrateEnum = {
    form_count: 'form_count',
};
/**
 *
 */
exports.GetFlowsByIdHydrateEnum = {
    form_count: 'form_count',
};
/**
 *
 */
exports.GetFlowsExecutionsByExecutionIdHydrateEnum = {
    debug: 'debug',
};
/**
 *
 */
exports.GetFormsHydrateEnum = {
    flow_count: 'flow_count',
    links: 'links',
};
/**
 *
 */
exports.GetFormsByIdHydrateEnum = {
    flow_count: 'flow_count',
    links: 'links',
};
/**
 *
 */
exports.PutFactorsByNameOperationNameEnum = {
    push_notification: 'push-notification',
    sms: 'sms',
    email: 'email',
    duo: 'duo',
    otp: 'otp',
    webauthn_roaming: 'webauthn-roaming',
    webauthn_platform: 'webauthn-platform',
    recovery_code: 'recovery-code',
};
/**
 *
 */
exports.GetHooksTriggerIdEnum = {
    credentials_exchange: 'credentials-exchange',
    pre_user_registration: 'pre-user-registration',
    post_user_registration: 'post-user-registration',
    post_change_password: 'post-change-password',
    send_phone_message: 'send-phone-message',
};
/**
 *
 */
exports.GetCustomTextByLanguagePromptEnum = {
    login: 'login',
    login_id: 'login-id',
    login_password: 'login-password',
    login_passwordless: 'login-passwordless',
    login_email_verification: 'login-email-verification',
    signup: 'signup',
    signup_id: 'signup-id',
    signup_password: 'signup-password',
    phone_identifier_enrollment: 'phone-identifier-enrollment',
    phone_identifier_challenge: 'phone-identifier-challenge',
    reset_password: 'reset-password',
    consent: 'consent',
    logout: 'logout',
    mfa_push: 'mfa-push',
    mfa_otp: 'mfa-otp',
    mfa_voice: 'mfa-voice',
    mfa_phone: 'mfa-phone',
    mfa_webauthn: 'mfa-webauthn',
    mfa_sms: 'mfa-sms',
    mfa_email: 'mfa-email',
    mfa_recovery_code: 'mfa-recovery-code',
    mfa: 'mfa',
    status: 'status',
    device_flow: 'device-flow',
    email_verification: 'email-verification',
    email_otp_challenge: 'email-otp-challenge',
    organizations: 'organizations',
    invitation: 'invitation',
    common: 'common',
    passkeys: 'passkeys',
};
/**
 *
 */
exports.GetCustomTextByLanguageLanguageEnum = {
    ar: 'ar',
    bg: 'bg',
    bs: 'bs',
    ca_ES: 'ca-ES',
    cs: 'cs',
    cy: 'cy',
    da: 'da',
    de: 'de',
    el: 'el',
    en: 'en',
    es: 'es',
    et: 'et',
    eu_ES: 'eu-ES',
    fi: 'fi',
    fr: 'fr',
    fr_CA: 'fr-CA',
    fr_FR: 'fr-FR',
    gl_ES: 'gl-ES',
    he: 'he',
    hi: 'hi',
    hr: 'hr',
    hu: 'hu',
    id: 'id',
    is: 'is',
    it: 'it',
    ja: 'ja',
    ko: 'ko',
    lt: 'lt',
    lv: 'lv',
    nb: 'nb',
    nl: 'nl',
    nn: 'nn',
    no: 'no',
    pl: 'pl',
    pt: 'pt',
    pt_BR: 'pt-BR',
    pt_PT: 'pt-PT',
    ro: 'ro',
    ru: 'ru',
    sk: 'sk',
    sl: 'sl',
    sr: 'sr',
    sv: 'sv',
    th: 'th',
    tr: 'tr',
    uk: 'uk',
    vi: 'vi',
    zh_CN: 'zh-CN',
    zh_TW: 'zh-TW',
};
/**
 *
 */
exports.GetPartialsPromptEnum = {
    login: 'login',
    login_id: 'login-id',
    login_password: 'login-password',
    login_passwordless: 'login-passwordless',
    signup: 'signup',
    signup_id: 'signup-id',
    signup_password: 'signup-password',
    customized_consent: 'customized-consent',
};
/**
 *
 */
exports.PutCustomTextByLanguagePromptEnum = {
    login: 'login',
    login_id: 'login-id',
    login_password: 'login-password',
    login_passwordless: 'login-passwordless',
    login_email_verification: 'login-email-verification',
    signup: 'signup',
    signup_id: 'signup-id',
    signup_password: 'signup-password',
    phone_identifier_enrollment: 'phone-identifier-enrollment',
    phone_identifier_challenge: 'phone-identifier-challenge',
    reset_password: 'reset-password',
    consent: 'consent',
    logout: 'logout',
    mfa_push: 'mfa-push',
    mfa_otp: 'mfa-otp',
    mfa_voice: 'mfa-voice',
    mfa_phone: 'mfa-phone',
    mfa_webauthn: 'mfa-webauthn',
    mfa_sms: 'mfa-sms',
    mfa_email: 'mfa-email',
    mfa_recovery_code: 'mfa-recovery-code',
    mfa: 'mfa',
    status: 'status',
    device_flow: 'device-flow',
    email_verification: 'email-verification',
    email_otp_challenge: 'email-otp-challenge',
    organizations: 'organizations',
    invitation: 'invitation',
    common: 'common',
    passkeys: 'passkeys',
};
/**
 *
 */
exports.PutCustomTextByLanguageLanguageEnum = {
    ar: 'ar',
    bg: 'bg',
    bs: 'bs',
    ca_ES: 'ca-ES',
    cs: 'cs',
    cy: 'cy',
    da: 'da',
    de: 'de',
    el: 'el',
    en: 'en',
    es: 'es',
    et: 'et',
    eu_ES: 'eu-ES',
    fi: 'fi',
    fr: 'fr',
    fr_CA: 'fr-CA',
    fr_FR: 'fr-FR',
    gl_ES: 'gl-ES',
    he: 'he',
    hi: 'hi',
    hr: 'hr',
    hu: 'hu',
    id: 'id',
    is: 'is',
    it: 'it',
    ja: 'ja',
    ko: 'ko',
    lt: 'lt',
    lv: 'lv',
    nb: 'nb',
    nl: 'nl',
    nn: 'nn',
    no: 'no',
    pl: 'pl',
    pt: 'pt',
    pt_BR: 'pt-BR',
    pt_PT: 'pt-PT',
    ro: 'ro',
    ru: 'ru',
    sk: 'sk',
    sl: 'sl',
    sr: 'sr',
    sv: 'sv',
    th: 'th',
    tr: 'tr',
    uk: 'uk',
    vi: 'vi',
    zh_CN: 'zh-CN',
    zh_TW: 'zh-TW',
};
/**
 *
 */
exports.PutPartialsPromptEnum = {
    login: 'login',
    login_id: 'login-id',
    login_password: 'login-password',
    login_passwordless: 'login-passwordless',
    signup: 'signup',
    signup_id: 'signup-id',
    signup_password: 'signup-password',
    customized_consent: 'customized-consent',
};
/**
 *
 */
exports.GetSelfServiceProfileCustomTextLanguageEnum = {
    en: 'en',
};
/**
 *
 */
exports.GetSelfServiceProfileCustomTextPageEnum = {
    get_started: 'get-started',
};
/**
 *
 */
exports.PutSelfServiceProfileCustomTextLanguageEnum = {
    en: 'en',
};
/**
 *
 */
exports.PutSelfServiceProfileCustomTextPageEnum = {
    get_started: 'get-started',
};
/**
 *
 */
exports.DeleteMultifactorByProviderProviderEnum = {
    duo: 'duo',
    google_authenticator: 'google-authenticator',
};
/**
 *
 */
exports.DeleteUserIdentityByUserIdProviderEnum = {
    ad: 'ad',
    adfs: 'adfs',
    amazon: 'amazon',
    apple: 'apple',
    dropbox: 'dropbox',
    bitbucket: 'bitbucket',
    aol: 'aol',
    auth0_oidc: 'auth0-oidc',
    auth0: 'auth0',
    baidu: 'baidu',
    bitly: 'bitly',
    box: 'box',
    custom: 'custom',
    daccount: 'daccount',
    dwolla: 'dwolla',
    email: 'email',
    evernote_sandbox: 'evernote-sandbox',
    evernote: 'evernote',
    exact: 'exact',
    facebook: 'facebook',
    fitbit: 'fitbit',
    flickr: 'flickr',
    github: 'github',
    google_apps: 'google-apps',
    google_oauth2: 'google-oauth2',
    instagram: 'instagram',
    ip: 'ip',
    line: 'line',
    linkedin: 'linkedin',
    miicard: 'miicard',
    oauth1: 'oauth1',
    oauth2: 'oauth2',
    office365: 'office365',
    oidc: 'oidc',
    okta: 'okta',
    paypal: 'paypal',
    paypal_sandbox: 'paypal-sandbox',
    pingfederate: 'pingfederate',
    planningcenter: 'planningcenter',
    renren: 'renren',
    salesforce_community: 'salesforce-community',
    salesforce_sandbox: 'salesforce-sandbox',
    salesforce: 'salesforce',
    samlp: 'samlp',
    sharepoint: 'sharepoint',
    shopify: 'shopify',
    sms: 'sms',
    soundcloud: 'soundcloud',
    thecity_sandbox: 'thecity-sandbox',
    thecity: 'thecity',
    thirtysevensignals: 'thirtysevensignals',
    twitter: 'twitter',
    untappd: 'untappd',
    vkontakte: 'vkontakte',
    waad: 'waad',
    weibo: 'weibo',
    windowslive: 'windowslive',
    wordpress: 'wordpress',
    yahoo: 'yahoo',
    yammer: 'yammer',
    yandex: 'yandex',
};
/**
 *
 */
exports.GetUsersSearchEngineEnum = {
    v1: 'v1',
    v2: 'v2',
    v3: 'v3',
};
